src/core/components/files-regular/add-from-url.js

```diff
@@ -14,7 +14,10 @@ module.exports = (self) => {
 
     try {
       const parsedUrl = new URL(url)
```

Comment:
this is need for the 
`const path = decodeURIComponent(parsedUrl.pathname.split('/').pop())`
below

we should have the url in the response already in the form of an URL instance and there is no need to `decodeURIComponent` can you check that please ?